File: Program diferents lenguages/Builder/HouseBuilderJava/HouseBuilder/src/HouseBuilder/BasicHouseBuilder.java

package HouseBuilder;

public class BasicHouseBuilder implements HouseBuilder{

	private House house;
	
	public BasicHouseBuilder() {
		this.house = new House();
	}
	
	@Override
	public void buildBasement() {
		house.setBasement("Unfurnished");
	}

	@Override
	public void buildRoof() {
		house.setRoof("Metal Roof");
	}

	@Override
	public void buildStructure() {
		house.setStructure("Minimal");
	}

	@Override
	public void buildInterior() {
		house.setInterior("Average");
	}
	
	public House getHouse() {
		return this.house;
		
	}

}

File: Program diferents lenguages/Builder/HouseBuilderC#/HouseBuilder/BasicHouseBuilder.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace HouseBuilder
{
    class BasicHouseBuilder : HouseBuilder
    {
        private House house;

        public BasicHouseBuilder()
        {
            this.house = new House();  //composition
        }


        public void buildBasement()
        {
            house.setBasement("unfinished");
        }

        public void buildStructure()
        {
            house.setStructure("Minimalist");
        }

        public void buildRoof()
        {
            house.setRoof("Metal Roofing");
        }

        public void buildInterior()
        {
            house.setInterior("good");
        }

        public House getHouse()
        {
            return this.house;
        }

    }
}
